### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by tank1610**
Logging functionality for the board.

### CODE REVIEW

**Comment by tank1610** [comment]
Calling log_sensor() will write a 24-byte log entry to the flash. Calling log_read_all() will read the content of the flash and print it over serial.
To be done: create an offload function on board and PC to store the log data when we enter safe mode.

**Comment by j-levy** [comment]
I can't see anywhere where we actually check what's written. At the moment, we only assume that what's written is written correctly, right ?

I'm not against merging, but I just want to be sure of what is done, and what is still needed.. We will have to check later on if the writing is done correctly with appropriate offload and other stuff, am I right ?

File: software_package/in4073/in4073.c [hunk 1]
```diff
@@ -192,6 +193,9 @@ int main(void)
 		{
 			tx_timer = get_time_us();
 			send_telemetry_packet();
+			log_write_item();
```
Comment (niketagrawal) [inline software_package/in4073/in4073.c:305]: log_write_item() should be replaced by log_sensor()?

File: software_package/in4073/telemetry_tx.c [hunk 2]
```diff
@@ -45,10 +45,5 @@ void send_telemetry_packet()
 		//fprintf(stderr, "\n");
         printf("\n");
 	#endif
-    //reset packet	
-	for (int j = 0; j < TELEMETRY_PACKET_SIZE; j++)
```
Comment (niketagrawal) [inline software_package/in4073/telemetry_tx.c:49]: since you are not using the log_write_item method, this should be uncommented now?

**Review by niketagrawal — COMMENTED** [review]
Current mode, incoming joystick/keyboard data and motor speeds also needs to be stored in the logged data as per requirements mentioned in the lab manual

File: software_package/in4073/telemetry_tx.c (hunk 2 above)
Comment (tank1610) [reply software_package/in4073/telemetry_tx.c:49]: That's correct. My bad

**Comment by tank1610** [comment]
I've checked it using the log_read_all() function and it seems fine. But do you mean checking if the log is written correctly during operation? If so then no, there isn't such a check. The write function returns false if it fails to write. If it succeeds then I assume it's written correctly. If an error or two occur somewhere it shouldn't be a big problem. 
Indeed, offloading to PC is still to be done.

**Pull request merged**